eZ/Publish/Core/REST/Server/Controller/Location.php

```diff
@@ -245,7 +246,7 @@ public function moveSubtree($locationPath, Request $request)
                 // Trash the subtree
                 $trashItem = $this->trashService->trash($locationToMove);
 
-                if (isset($trashItem)) {
+                if ($trashItem instanceof TrashItem) {
```

Comment:
Ok, this answers my previous question. 
So this could be 
```php 
if ($trashItem !== null && $trashItem->contentInfo !== null)
```

Reply:
No longer relevant, using sudo instead.